pkg/prometheus/promcfg.go

```diff
@@ -1804,10 +1804,22 @@ func (l *limitEnforcer) addBodySizeLimitsToYAML(cfg yaml.MapSlice, enforcedLimit
 		return cfg
 	}
 
+	if !validInput("^(0[K,M,G,T,P,E]*B$|[1-9]\\d+[K,M,G,T,P,E]*B$)", enforcedLimit) {
```

Comment:
Should we use the same library as prometheus [1][2] instead of crafting our own regex? 

[1] https://github.com/prometheus/prometheus/blob/e582e907f8a97ffae27123b9632a254e5018a4a6/config/config.go#L400
[2] https://pkg.go.dev/github.com/alecthomas/units

Reply:
Yep using the same library makes sense.

Reply:
I initially created code using units but I dropped that since other fields also needed to be validated even though in this PR only bodysizelimit is validated. So thought to have a general validator method.

Reply:
https://github.com/prometheus-operator/prometheus-operator/compare/master...slashpai:input_validator#diff-453ce9c721c51bf9591c2b212e9ce50157b9c40d1b211177bf502be268099656R1808-R1816

Reply:
I think a general validator can be useful as a fallback when more specific ones are not available. Regular expressions can often be hard to verify and troubleshoot, so I would keep things simple if possible.

Reply:
Sure that makes sense, I will change back the validation using units